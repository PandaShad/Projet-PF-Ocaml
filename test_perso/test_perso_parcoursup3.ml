open Test_perso_parcoursup_utils

let session = nouvelle_session ()

let () =
    ajoute_candidat session ~nom_candidat:"Adam";
    ajoute_candidat session ~nom_candidat:"Marie";
    ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:2;
    ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:1;
    ajoute_voeu session
        ~rang_repondeur:(Some 1)
        ~nom_candidat:"Adam"
        ~nom_formation:"Université Côte d'Azur";
    ajoute_voeu session
        ~rang_repondeur:(Some 0)
        ~nom_candidat:"Adam"
        ~nom_formation:"Université de Toulon";
    ajoute_voeu session
        ~rang_repondeur:(Some 1)
        ~nom_candidat:"Marie"
        ~nom_formation:"Université Côte d'Azur";

    ajoute_commission session
        ~nom_formation:"Université Côte d'Azur"
        ["Adam";"Marie"];
    ajoute_commission session
        ~nom_formation:"Université de Toulon"
        ["Adam";"Marie"];
    reunit_commissions session;

    nouveau_jour session;
    affiche_voeux_en_attente session "Adam";
    affiche_propositions_en_attente session "Adam";
    affiche_voeux_en_attente session "Marie";
    affiche_propositions_en_attente session "Marie";

    renonce session
        ~nom_candidat:"Adam"
        ~nom_formation:"Université de Toulon";

    nouveau_jour session;
    affiche_voeux_en_attente session "Adam";
    affiche_propositions_en_attente session "Adam";
    affiche_voeux_en_attente session "Marie";
    affiche_propositions_en_attente session "Marie";