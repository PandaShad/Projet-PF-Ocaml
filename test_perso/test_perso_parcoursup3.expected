nouvelle session
ajoute candidat Adam
ajoute candidat Marie
ajoute formation Université de Toulon (2 places)
ajoute formation Université Côte d'Azur (1 places)
ajoute voeu de Adam pour Université Côte d'Azur (rang=1)
ajoute voeu de Adam pour Université de Toulon (rang=0)
ajoute voeu de Marie pour Université Côte d'Azur (rang=1)
commission de classement de Université Côte d'Azur : [Adam; Marie]
commission de classement de Université de Toulon : [Adam; Marie]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université de Toulon}
Voeux en attente de Marie : {}
Propositions en attente de Marie : {Université Côte d'Azur}
Adam renonce à Université de Toulon
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {}
Voeux en attente de Marie : {}
Propositions en attente de Marie : {Université Côte d'Azur}
